evalml/pipelines/components/transformers/transformer.py

```diff
@@ -73,9 +69,8 @@ def fit_transform(self, X, y=None):
             y_ww = infer_feature_types(y)
         try:
             X_t = self._component_obj.fit_transform(X_ww, y_ww)
-            return _retain_custom_types_and_initalize_woodwork(
-                X_ww.ww.logical_types, X_t
-            )
+            X_t.ww.init(schema=X_ww.ww.schema)
```

Comment:
This could be a good location to use `X_t.ww.init_with_full_schema`, which is stricter than `init` in that it would require `X_t` to already be valid for the schema (meaning none of its dtypes need to be updated at init, for example). 

I'm not sure whether that's actually true here, but if it's something that you'd expect (say, none of the dtypes have changed from X_ww during the `fit_transform` call to create X_t), using `init_with_full_schema` will alert you to situations where that's not the case instead of silently re-updating the dtype.

In general, it might be worth looking over the places where you're using `init(schema=schema)` across evalML and changing to `init_with_full_schema` in any locations where you would expect the dataframe to already be valid for that schema and where it'd be problematic if it wasn't.